src/cookieutils.ts

```diff
@@ -53,8 +30,7 @@ export class Cookie {
         var cookieArray = document.cookie.split(';');
         for (var i = 0; i < cookieArray.length; i++) {
             var cookie = cookieArray[i];
-            cookie = cookie.replace(/^\s+/, '');
-
+            cookie = cookie.replace(/^\s+/, ''); //strip whitespace from front of cookie only
             if (cookie.indexOf(cookiePrefix) == 0) {
```

Comment:
This code has written out [startsWith](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith), but poorly. The "poor man's version" should use `.lastIndexOf(prefix, 0)` to avoid searching the whole string for no good reason at all. However, since support for `startsWith` is so ubiquitous, I suggest:
```suggestion
            if (cookie.startsWith(cookiePrefix)) {
```